pkgs/by-name/ti/tic-80/package.nix

```diff
@@ -0,0 +1,143 @@
+{ lib
+, pkgs
+, stdenv
+, fetchFromGitHub
+, libGL
+, alsa-lib
+, libX11
+, xorgproto
+, libICE
+, libXi
+, libXScrnSaver
+, libXcursor
+, libXinerama
+, libXext
+, libXxf86vm
+, libXrandr
+, libxkbcommon
+, wayland
+, wayland-protocols
+, wayland-scanner
+, dbus
+, udev
+, libdecor
+, pipewire
+, libpulseaudio
+, libiconv
+, ...
+}:
+with pkgs;
+stdenv.mkDerivation rec {
+  pname = "tic-80";
+  version = "1.1.2837";
+
+  src = fetchFromGitHub {
+    owner = "nesbox";
+    repo = "TIC-80";
+    rev = "v" + version;
+    sha256 = "sha256-2hSaLWw57F19tSfgJvBgbqW52vCu8p/TmoybYzQEybE=";
+    fetchSubmodules = true;
```

Comment:
Do we really, really need to use that many vendored libraries?

Reply:
It also make me a bit sad. [aseprite](https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/editors/aseprite/default.nix) is another program with lost of vendored dependencies, but it also offers to use the shared libraries from the environment via CMake options like `-DUSE_SHARED_ZLIB=ON`. I'd like that for TIC-80, as well, but it'd be work to include it upstream.

So I'd propose to do it as a multistep process, where we solve it like this for now, and then go ahead and bring a more flexible build process to TIC-80.

Reply:
Could you add that limitation as a comment?